### DISCUSSION TITLE
There are no commands defined in the "firefly-iii" namespace.

### DISCUSSION THREAD

**Discussion by Terry-JF** [firefly-iii/firefly-iii#9071] [Firefly III] [answered]
I am trying to run the data export routines using the command 'php artisan firefly-iii:export-data .....' but irrespective of which command I enter I always get the same error:

"There are no commands defined in the "firefly-iii" namespace."

I found a reference to a similar problem in issue #3113 from back in February 2020 and it seems that further development was required to provide these commands at that point. Is this still the case or am I doing something wrong?

Thanks,
Terry

**Comment by JC5** [comment]
No the command should be right there, together with another 30-something commands in the "firefly-iii" namespace.

Could you tell me a little bit more about your setup?

**Reply by Terry-JF** [reply]
I have a self-built Linux Mint running on a MacMini 6,2 with FF3 in /var/www/html/firefly-iii and data-importer in /var/www/html/data-importer. (No Docker - I'd never heard of it until 3 weeks ago!). Command executed was 'sudo php artisan firefly-iii:export-data --export-categories'

Just realised that the terminal prompt was located in' /var/www/html/data-importer' when I tried this. So re-tried from '/var/www/html/firefly-iii' and got a different error:

[2024-07-22 11:46:15] production.ERROR: Invalid access token for user #1.  
[2024-07-22 11:46:15] production.ERROR: Token given is "", expected something else.  
  [x]  Invalid access token. Check /profile.

Sorry if this is noob stuff, learning as I go. On the bright side, the auto-import is working great, cron job is pulling transactions from 7 different banks every day into FF3 and then my personal home-brew accounts system (Windows based) uses the API to get a copy of these, (it mainly does forward financial planning - lots of forecast transactions in there into 2026). Thanks for this.

**Reply by JC5** [reply, marked as answer]
Great! Not a problem of course!

The token you need can be found on the following page: `/profile`. Click the "command line token" tab.

The parameter you need to add is `--token=tokenhere`. If your user ID is not 1, you also need `--user=x`.

**Reply by Terry-JF** [reply]
Excellent James - worked like a charm.

Very many thanks.
